### DISCUSSION TITLE
I'd like to update the URL slug (not the query string)

### DISCUSSION THREAD

**Discussion by IvaWonderbush** [livewire/livewire#1792] [4. General]
When on the route `/post` I display a form to create a new post. Once saved, I would like to change the URL to `/post/{post_id}`. Example: `/post/12`

I can solve this with my own custom emit for now, but it would be cool if this was built in!

`$this->emit('slug', route('post.create', 12))` or something :)

**Comment by nuernbergerA** [comment]
What's wrong with a normal [redirect](https://laravel-livewire.com/docs/2.x/redirecting)?

There is also `$this->redirectRoute()` and `$this->redirectAction()`

**Reply by IvaWonderbush** [reply]
It refreshes the page. It feels a bit weird when the validation and submitting the form doesn't perform a refresh. Or am I doing something wrong? (I've had a few issues with Livewire so far, so it could be that it's just me).

**Reply by nuernbergerA** [reply]
Maybe you can explain your use-case a bit more in detail?

**Reply by IvaWonderbush** [reply]
Oh sure. I have a form (with real time validation) at the route `/post`, and when you submit the form you create a new post. I save the newly created ID so you can directly edit the post you just created, but I would like the URL to reflect the newly created post. You can edit a post by using its specific ID `/post/12` (12 is an example ID). Seeing as validation and submitting the form is all real time with no refresh I'd like to change the URL to `/post/12` once you create a new post as well without a refresh.
